app/store/providers/data.tsx

```diff
@@ -0,0 +1,33 @@
+'use client';
+
+import React, { createContext, useContext, useState, ReactNode } from 'react';
+import { ReactQueryProvider } from "@developmentseed/veda-ui";
+
+interface DataStore {
+  datasets?: any[];
+  setDatasets?: React.Dispatch<React.SetStateAction<any[] | undefined>>;    
+};
+
+export const DataContext = createContext<DataStore>({});
+
+export function useDataStore() {
+  return useContext<DataStore>(DataContext);
+}
+
+function DataProvider({ initialDatasets = undefined, children }: {children: JSX.Element | ReactNode, initialDatasets: any[] | undefined}) {
```

Comment:
Oh, I might have a clue about why all the contents are in Javscript, but not statically generated. 

We are saving all the data related to a dataset in Provider here. However, other than the dataset overview page, the contents for the datasets are not necessary throughout the app. If we save the contents in Provider, the data that Provider needs to hold is way too big - and the pages that don't need the content have to suffer from loading it.  (And I am guessing this is why VEDA faux module didn't include the contents as a part of Dataset).

Can we save the metadata in Provider, but not the contents?  (Then Dataset page component would be in charge of loading the content I think?)

Reply:
Hmm i'm not too sure I understand your concern here 🤔. So how this is working is that we initialize with the datasets to the provider at the highest level. The provider is just a global state store. Pages dont need to worry about loading/rendering it unless we use the store (context). This is an ideal/standard pattern too. The only longer load time is when we first initialize but that should be fine in my opinion.

Reply:
Ah! I am sorry, I was mixing two separate problems.

First - the problem is that the page doesn't render the contents. It looks like we need to add a line for configuration: https://nextjs.org/docs/pages/building-your-application/deploying/static-exports#configuration (And we will need to change some parts of the code too, so let's separate this problem for now.)

I mistook that this PR uses the provider to pass the list of the datasets to generate the pages, but I see that the pages are generated with raw mdx util function: https://github.com/developmentseed/next-veda-ui/pull/1/files#diff-391fde509832e41741dd8407f511821a199808eeedf0c29ee27889326c542b45R55  My apologies!  Then two questions here.

1. Can we just use `getDatasets()` instead of `useDatastore` here? https://github.com/developmentseed/next-veda-ui/pull/1/files#diff-0e88e4f8aa269c374b38ecd19c41262467a171e8562e52e2f648f2a8da89122dR7

2. Do we need the content data in a global state? https://github.com/developmentseed/next-veda-ui/pull/1/files#diff-d09c73561249208ba27367f1521580430f73d736ddb203c3f78e930725d89548R38

Reply:
Spoke with @hanbyul-here: 
* Do not load content into data provider - fetch by data overview page when clicking into a dataset
* Look into wrapping providers for specific routes because not every route needs the datasets or stories